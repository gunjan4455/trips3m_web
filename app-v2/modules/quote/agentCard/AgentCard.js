import React from 'react';
import PropTypes from 'prop-types';
import Sensor from 'react-visibility-sensor';
import { Link } from 'react-router-dom';

import { CallUs, ChatIcon } from 'helpers/Icon/Icon';
import { ORGANISM_CLICKED } from 'actions/segmentEvents';
import RequestCallback from 'modules/requestCallback';
import AgentChatButton from 'modules/quote/shared/AgentChatButton';
import BasicInfo from './components/BasicInfo';
import DiscountOffer from 'modules/discountEngine';
import AgentDetail from './components/AgentDetail';
import QuoteBottom from 'modules/quote/bottom';
import DealPrice from 'modules/quote/dealPrice';
import { ORGANISM_VIEWED } from 'actions/segmentEvents';
import { checkIsTripStatusBeforeInvoiced } from 'helpers/utils';

const AgentCard = ({
                     tripId, quoteId, agent, userDetails, destination, tripStatus, priceInformation,
                     discountInformation, downloadQuote, showRoster, trackSegment
                   }) => {

  const trackSegmentEvent = (event, section, object, cta) => {
    trackSegment({
      event,
      section: section || 'Agent',
      object: object || `Agent/${agent.id}`,
      cta
    });
  };

  const onSensorChange = (isVisible) => {
    if(isVisible) {
      trackSegment({ event: ORGANISM_VIEWED, section: 'Price Matching' });
    }
  };

  const hasDiscountApplied = discountInformation.discount > 0;
  const isTripStatusBeforeInvoiced = checkIsTripStatusBeforeInvoiced(tripStatus);
  return (<div className="sbcw mt8 mb2">
      <div className="p15 at_agentDetailCard">
        <AgentDetail userDetails={userDetails} agent={agent} />
        <div className="mt15 flex alignCenter spaceBetween at_agentActionCTA_section">
          <a href={`tel:${agent.phoneNumber}`} onClick={() => trackSegmentEvent(ORGANISM_CLICKED, 'Agent', '', 'Call')}
             className="iblock flex alignCenter pfc3 fw7 at_call_CTA">
            <span className="input-24 mr8"><CallUs /></span><span className="pfc3">Call</span>
          </a>
          <RequestCallback />
          <Link to="" className="iblock flex alignCenter pfc3 fw7 at_message_CTA">
            <span className="input-24 mr8"><ChatIcon /></span>
            <AgentChatButton textClassName={'pfc3, fw7'} name="Message"
                             handleChatClick={() => trackSegmentEvent(ORGANISM_CLICKED, 'Agent', '', 'Message')} />
          </Link>
        </div>
      </div>
      <BasicInfo tripId={tripId} quoteId={quoteId} destination={destination}
                 priceInformation={priceInformation} downloadQuote={downloadQuote}
                 discountInformation={discountInformation} showRoster={showRoster}
                 tripStatus={tripStatus} trackSegment={trackSegmentEvent} />

      <QuoteBottom.container trackSegment={trackSegment}/>
      <div className="pt8 sbc5" />
      {
        hasDiscountApplied
          ? <DiscountOffer discountInformation={discountInformation}
                           tripId={tripId}
                           renderAppliedCoupon={true} />
          : null
      }
    { isTripStatusBeforeInvoiced &&
      <Sensor onChange={(isVisible) => onSensorChange(isVisible) }
              partialVisibility='bottom'
              delayedCall={true}>
        <DealPrice trackSegment={trackSegmentEvent}/>
      </Sensor>}
    </div>
  );
};

AgentCard.propTypes = {
  tripId: PropTypes.number.isRequired,
  quoteId: PropTypes.number.isRequired,
  agent: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    tripsSold: PropTypes.number.isRequired,
    email: PropTypes.string.isRequired,
    phoneNumber: PropTypes.string.isRequired,
    imageUrl: PropTypes.string.isRequired,
    star: PropTypes.number,
    reviewsCount: PropTypes.number,
    reviewLink: PropTypes.string
  }),
  userDetails: PropTypes.object.isRequired,
  destination: PropTypes.string.isRequired,
  tripStatus: PropTypes.string.isRequired,
  priceInformation: PropTypes.object.isRequired,
  discountInformation: PropTypes.object,
  downloadQuote: PropTypes.func.isRequired,
  showRoster: PropTypes.func,
  trackSegment: PropTypes.func
};

AgentCard.defaultProps = {
  agent: {
    star: 0,
    reviewsCount: 0,
    reviewLink: ''
  },
  discountInformation: {},
  showRoster: () => {},
  trackSegment: () => {}
};

export default AgentCard;
