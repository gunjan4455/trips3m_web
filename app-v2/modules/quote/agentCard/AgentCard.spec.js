import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import AgentCard from './AgentCard';

configure({ adapter: new Adapter() });
let component;

const agentCardData = {
  tripId: 3918699,
  quoteId: 7124216,
  agent: {
    id: 1699179,
    name: 'Unique Trip Maker ',
    imageUrl: 'https://res.cloudinary.com/dma3zc8eo/image/fetch/f_auto,q_auto:eco,w_300/https://dknfdeb8v69ms.cloudfront.net/attachments/tr_a_profiles/3048/original/logo.png',
    star: 4,
    tripsSold: 298,
    reviewsCount: 97,
    reviewLink: 'http://bookings-qa1.ttdev.in/agents/313639393137392d556e69-Unique-Trip-Maker',
    email: 'info@uniquetripmaker.com',
    phoneNumber: '8860670640'
  },
  userDetails: {
    id: '112233'
  },
  destination: "Goa",
  priceInformation: {
    price: 7499,
    priceInRupee: 7499,
    currency: 'Rupee',
    perPerson: 'per person'
  },
  downloadUrl: 'https://bookings-qa1.ttdev.in/quotes/7124217/download_quote.pdf',
  discountInformation: {
    discount: 0,
    discountableTypeId: null,
    discountableType: null
  }
};

const createAgentCardComponent = () => {
  return (<AgentCard {...agentCardData} />);
};

describe('Agent Card: Render Correctly ', () => {

  beforeAll(() => {
    const AgentCardComponent = createAgentCardComponent();
    component = shallow(AgentCardComponent);
  });

  it('should match with the previous snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
