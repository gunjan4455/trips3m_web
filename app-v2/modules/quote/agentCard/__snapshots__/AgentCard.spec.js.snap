// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Agent Card: Render Correctly  should match with the previous snapshot 1`] = `undefined`;
